packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -436,8 +435,11 @@ function i(
                 // Check that the child vnode is either an element or VStatic
                 if (!isNull(childVnode) && (isVBaseElement(childVnode) || isVStatic(childVnode))) {
                     const { key } = childVnode;
+                    // In @lwc/engine-server the fragment doesn't have a tagName, default to the VM's tagName.
                     const tagName =
-                        childVnode.sel ?? StringToLowerCase.call(childVnode.fragment.tagName);
+                        childVnode.sel ??
+                        childVnode.fragment.tagName?.toLowerCase() ??
+                        vmBeingRendered.tagName;
```

Comment:
@nolanlawson I know we have the `stringToLowerCase` helper but it seemed easier to just use optional chaining here.

Let me know what you think, I can change it to use `stringToLowerCase` instead.

Reply:
We haven't been as consistent as we used to be, but the idea behind using cached functions is that we're less susceptible to prototype pollution.

Reply:
Sounds good, I updated to use the cached function in the latest commit - https://github.com/salesforce/lwc/pull/4103/commits/3dfa572f53816646af94645782cd6a44735ba622